### ISSUE TITLE
Expose Application.handleError

### ISSUE THREAD

**Issue by bbarkley** [playframework/playframework#2416]
Can Application.handleError be changed from private[play] to part of the public API?

We have code in a filter that needs to set a cookie even if the Controller or an upstream handler throws an exception. We'd like to keep the error handling behavior as close to the normal path as possible, but we need to have a SimpleResult to attache the cookie to. The basic code looks like:

```
for {
  result <- next(request).recoverWith {
    // recovery needed so we are able to set the cookie
    // would rather call app.handleError(req, t) though
    case t: Throwable => app.global.onError(req, t)
  }
  cookie <- buildCookie()
} yield {
  result.withCookies(cookie)
}
```

Since Application.handleError does the actual logging and a few other things we aren't getting the same behavior when just calling the global onError.

Is there a reason Application.handleError is private, or can it be exposed (or refactored to a helper object) for outside use?

**Comment by wsargent** [comment]
I think that with HttpRequstHandler and HttpErrorHandler, this should be less of an issue.

**Comment by wsargent** [comment]
This is fixed in 2.5.x because you can extend `DefaultHttpErrorHandler.onServerError` in your own code without calling out to global state:

https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/http/HttpErrorHandler.scala#L193
